### PR TITLE
Feature/us17526 add mobile header

### PR DESCRIPTION
**Pull request by silasbaur**

### CODE REVIEW

File: _assets/stylesheets/_header.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.navbar-default {
+  background-color: $cr-white;
+  margin: 0 auto;
+  z-index: 100;
+  
+  ul li a {
+    color: $cr-gray-dark !important;
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: I wouldn't expect to need an important tag here if we're not importing header styles from crds-styles.

File: _assets/stylesheets/_header.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    &:hover {
+      color: $cr-gray-light !important;
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: And here.

File: _assets/stylesheets/_header.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    float: left;
+    padding: 10px;
+    margin-left: 15px;
+    margin-right: 0px;
+
+    &:hover {
+      background: none;
+    }
+  }
+  .navbar-brand {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: Let's try to stay consistent with spacing. It's not a big deal, but looks like you've started the trend of adding a newline before a new selector, so let's stick with that.

File: _assets/stylesheets/_header.scss [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    width: 240px;
+
+    img {
+      padding: 0;
+    }
+  }
+  button {
+    cursor: pointer;
+  }
+  .mobile-home-nav {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: This naming feels a little weird to me. "mobile-home-nav"? Is this only for the home page?

File: _assets/stylesheets/_header.scss [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    display: none;
+  }
+}
+
+@media screen and (max-width : $grid-float-breakpoint) {
+
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: But here, I wouldn't leave a newline.

Comment (seancdavis) [reply _assets/stylesheets/_header.scss]: Also I'd go `max-width: $...` (no space between colon and "max-width")

File: _assets/stylesheets/_header.scss [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  .navbar-default .navbar-nav>li>a {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: We usually don't use spaces between the direct child selectors.

This also feels specific and generic at the same time. We try to avoid using generic tag selectors (`li`, `a`) and stick with classes. It's easier to control and override.

Personally, I try to be as flat as possible. I'll share more information about that with you.

File: _includes/_header.html [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#myNavbar" aria-expanded="false">
+          <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span>
+        </button>
+        <a href="/" class="navbar-brand mobile-soft-left" routerlink="/" ng-reflect-router-link="/" ng-reflect-href="/">
+          <img class="logo" src="/img/crds-music-logo.png" alt="Crossroads Music Logo">
+        </a>
+      </div>
+      <div class="collapse navbar-collapse" id="myNavbar">
+        <ul class="nav navbar-nav navbar-right">
+          <li><a href="/index" class="flush-ends mobile-home-link">Home</a></li>
```
Comment (seancdavis) [inline _includes/_header.html]: Home should link to `/`.

File: _includes/_header.html [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+          <li><a href="#" class="flush-ends">Albums &amp; Songs</a></li>
+          <li><a href="#" class="flush-ends">Listen anywhere</a></li>
+          <li><a href="/about" class="flush-ends">About</a></li>
+          <li><a href="/contact" class="flush-ends">Contact</a></li>
+          <li><button class="btn btn-gray-light" type="button">Subcribe</button></li>
```
Comment (seancdavis) [inline _includes/_header.html]: Something is wrong here... 😉

File: _assets/stylesheets/_header.scss (hunk 4 above)
Comment (silasbaur) [reply _assets/stylesheets/_header.scss]: This was a mistake. It is supposed to be mobile-home-link. This class is for home to show up in the mobile menu in the navbar.

File: _assets/stylesheets/_header.scss [hunk 9]
```diff
@@ -0,0 +1,114 @@
+.navbar-default {
+  background-color: $cr-white;
+  margin: 0 auto;
+  z-index: 100;
+  
+  ul li a {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: Why are we nested and so specific here? Couldn't we create a selector called `.navbar-link` (or similar), and leave the styles flat?

File: _assets/stylesheets/_header.scss [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+    &:hover {
+      background: none;
+    }
+  }
+
+  .navbar-brand {
+    width: 240px;
+
+    img {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: I know it seems like it's not a big deal, but let's try to make a habit out of using classnames and not generic selectors. It then enables us to make this file much flatter.

In other words, this could be something like `.navbar-brand-img`, and then could be moved away from its parent.

It seems silly, but we've ran into issues with generic selectors in the past, so I'd like to avoid them unless we're affecting the global element.

Comment (seancdavis) [reply _assets/stylesheets/_header.scss]: This applies to all generic selectors in this PR.

File: _assets/stylesheets/_header.scss [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+    font-size: 15px;
+    margin-left: 15px;
+  }
+  .mobile-home-link {
+    display: none;
+  }
+}
+
+@media screen and (max-width: $grid-float-breakpoint) {
+  .navbar-default .navbar-nav .nav-link {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: Why so specific?

Comment (seancdavis) [reply _assets/stylesheets/_header.scss]: There are a few others below that also seem super specific.

File: _assets/stylesheets/_header.scss [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    cursor: pointer;
+    font-size: 15px;
+    margin-left: 15px;
+  }
+  .mobile-home-link {
+    display: none;
+  }
+}
+
+@media screen and (max-width: $grid-float-breakpoint) {
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: In sass, the best practice we follow is that media queries go _within_ the rule in which they are affecting. That way we keep selectors grouped together.

File: _assets/stylesheets/_header.scss [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+  .navbar-default .navbar-nav .nav-link {
+    border-bottom: 1px solid $cr-gray-lighter;
+    font-size: 15px;
+  }
+
+  .navbar .navbar-nav ul li {
+    position: absolute;
+    z-index: 1000;
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: Why this number and why is this necessary?

File: _assets/stylesheets/_header.scss [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+    display: block;
+    margin: 10px auto;
+    width: 75%;
+  }
+
+  .navbar-brand {
+    transform: translateX(-50%);
+    left: 50%;
+    position: absolute;
+    margin: 0px !important;
```
Comment (seancdavis) [inline _assets/stylesheets/_header.scss]: Why?

Comment (seancdavis) [reply _assets/stylesheets/_header.scss]: Same question for all the other importants below.
